lib/galaxy/jobs/__init__.py

```diff
@@ -579,9 +588,7 @@ def __set_default_job_conf(self):
             self._set_default_handler_assignment_methods()
         else:
             self.app.application_stack.init_job_handling(self)
```

Comment:
these two lines moved to `__init__`

Reply:
I don't think that works ... it is definitely  a code smell and we should have this in `__init__`, but the handler setup is really complicated.

Reply:
> Should we create a ticket to refactor this?

Yes, please

Reply:
Done https://github.com/galaxyproject/galaxy/issues/12970